tests/stub/driver_parameters/test_connection_acquisition_timeout_ms.py

```diff
@@ -0,0 +1,116 @@
+            self._driver.close()
+
+        return super().tearDown()
+
+    def test_should_encompass_the_handshake_time(self):
+        self._server.start(
+            self.script_path("session_run_auth_delay.script")
+        )
+
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+        uri = "bolt://%s" % self._server.address
+        self._driver = Driver(self._backend, uri, auth,
+                              connection_acquisition_timeout_ms=2000,
+                              connection_timeout_ms=720000)
+
+        self._session = self._driver.session("w")
+
+        with self.assertRaises(types.DriverError):
+            list(self._session.run("RETURN 1 as n"))
+
+    def test_should_have_priotity_over_the_connection_timeout(self):
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+
+        # Non routable address
+        uri = "bolt://10.255.255.255"
+        self._driver = Driver(self._backend, uri, auth,
+                              connection_acquisition_timeout_ms=2000,
+                              connection_timeout_ms=720000)
+
+        self._session = self._driver.session("w")
+
+        with self.assertRaises(types.DriverError):
+            list(self._session.run("RETURN 1 as n"))
+
+    def test_should_not_suppress_connection_timeout(self):
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+
+        # Non routable address
+        uri = "bolt://10.255.255.255"
+        self._driver = Driver(self._backend, uri, auth,
+                              connection_acquisition_timeout_ms=72000,
+                              connection_timeout_ms=2000)
+
+        self._session = self._driver.session("w")
+
+        with self.assertRaises(types.DriverError):
+            list(self._session.run("RETURN 1 as n"))
+
+    @driver_feature(
+        types.Feature.TMP_DRIVER_MAX_CONNECTION_POOL_SIZE,
+        types.Feature.OPT_EAGER_TX_BEGIN
+    )
+    def test_should_regulates_the_time_for_acquiring_connections(self):
+        self._server.start(
+            self.script_path("tx_without_commit_or_rollback.script")
+        )
+
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+        uri = "bolt://%s" % self._server.address
+        self._driver = Driver(self._backend, uri, auth,
+                              connection_acquisition_timeout_ms=2000,
+                              connection_timeout_ms=720000,
+                              max_connection_pool_size=1)
+
+        self._sessions = [
+            self._driver.session("w"),
+            self._driver.session("w"),
+        ]
+
+        self._txs = [self._sessions[0].begin_transaction()]
+
+        with self.assertRaises(types.DriverError):
```

Comment:
just to be sure I'm following, this would fail after the acquisition timeout is reached before the active connection is freed up?

Reply:
Exactly. In this scenario, I've set up the connection pool size to `1` and blocked only available connection with a transaction.